core-bundle/contao/drivers/DC_Folder.php

```diff
@@ -1467,8 +1473,14 @@ public function edit()
 						$objFile = is_dir($this->strRootDir . '/' . $this->intId) ? new Folder($this->intId) : new File($this->intId);
 
 						$this->strPath = StringUtil::stripRootDir($objFile->dirname);
-						$this->strExtension = $objFile->origext ? '.' . $objFile->origext : '';
-						$this->varValue = $objFile->filename;
+
+						if (($strExtension = $objFile->origext) === 'twig')
+						{
+							$strExtension = ContaoTwigUtil::getExtension($objFile->name);
+						}
+
+						$this->strExtension = $strExtension ? '.' . $strExtension : '';
+						$this->varValue = substr($objFile->name, 0, -\strlen($this->strExtension));
```

Comment:
```suggestion
						$this->varValue = substr($objFile->name, 0, -\strlen($this->strExtension) ?: null);
```
Otherwise if `$this->strExtension === ''` the value would be set to the empty string